### DISCUSSION TITLE
Will not install on newish Fedora 36 install i get errors

### DISCUSSION THREAD

**Discussion by thorrrr** [abraunegg/onedrive#2085] [General]
HI Guys 
Just ditched Rclone for your app but soon as i try with your lateest rpm release i get an error!
Is it a bug or is there an easy fix?

 fedora 36 is only at 2.0.97 and onedrive needs 100.

https://i.imgur.com/EkooNkj.png

![image](https://user-images.githubusercontent.com/8170673/183752983-b1328c09-971d-448e-ab05-7edb26ac7154.png)

**Comment by abraunegg** [comment]
@thorrrr 
It appears you are trying to install the client on Fedora 37 ... which is the beta version of Fedora. You need to reconfigure your system to not use the Fedora 37 beta repositories.

To install correctly on Fedora 36:
* Install the Fedora 36 'onedrive' package - but this will only give you v2.4.19 which 1 revision behind the current release and is not supported
* Install the client from source following https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#dependencies-fedora--version-18--centos-8x--rhel-8x--rhel-9x

Additionally, I am not responsible for any 'packaging' issues - This is the wrong place to report them. Please report your packaging and dependency issue on https://bugzilla.redhat.com/buglist.cgi?quicksearch=onedrive

**Comment by thorrrr** [comment]
HI
I managed to get it working using Fedora Repositry
I did a dry run with no issues
I have created a sync list file 
But when  i go to resync i get an error
```
WARNING: Your onedrive client version is obsolete and unsupported. Please upgrade your client version.

Configuring Global Azure AD Endpoints

--synchronize or --monitor switches missing from your command line input. Please add one (not both) of these switches to your command line or use 'onedrive --help' for further assistance.
```
ALso on a side not i have Office 365 Family i guess i just use the Personal Method

**Reply by abraunegg** [reply]
@thorrrr 
Please read the usage document for assistance

**Reply by thorrrr** [reply]
Hello 
I run a dry run this morning i was running for hours in a loop doing the same stuff.
I have added this sync-list file again took 30 plus minutes looping same items 
But produces an error
``` 
ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 500 (Internal Server Error)
  Error Reason:     Failed to get a coherent view of changes. Try again later.
  Error Timestamp:  2022-08-10T09:30:12
  API Request ID:   bcf2573e-6485-4743-a752-3353203811a5
  Calling Function: applyDifferences()
```  
Not sure how reading which i have several times helps me!

**Comment by abraunegg** [comment]
@thorrrr 
> Not sure how reading which i have several times helps me!

This was in response to your usage error of trying to use `--resync` without using an applicable `--synchronize` or `--monitor` directive.

So to help you get started:
#### Step 1
Run `onedrive --display-config`

Ensure that the application is correctly configured as per your requirement.

Keep editing your configuration file and/or sync_list file until this command displays your desired configuration

#### Step 2
As you have configured 'sync_list' and edited it, you need to at least use `--resync` *once*, to flush your local cache.

Run `onedrive --synchronize --verbose --dry-run --resync --resync-auth`

As per the documentation, this will show you what would happen on a sync. No data is transferred, no local cache is updated.

#### Step 3
Remove the 'dry-run' condition

Run `onedrive --synchronize --verbose --resync --resync-auth`

This will now sync your data between your OneDrive account and update the local cache so that a record of what has been synced is available.

#### Step 4
Now that you are in a synchronised position, perform a sync but without the 'resync'

Run `onedrive --synchronize --verbose`

Validate that everything is syncronised via:

Run `onedrive --display-sync-status`

#### Step 5 (Optional - but no automatic sync will be performed without this)
Enable any systemd service following https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-as-a-system-service


Now .. as for this error:
>```
>ERROR: Microsoft OneDrive API returned an error with the following message:
>  Error Message:    HTTP request returned status code 500 (Internal Server Error)
>  Error Reason:     Failed to get a coherent view of changes. Try again later.
>  Error Timestamp:  2022-08-10T09:30:12
>  API Request ID:   bcf2573e-6485-4743-a752-3353203811a5
>  Calling Function: applyDifferences()
>```

This is a OneDrive API error - a 500 error is something failing on the Microsoft side ... nothing that I can assist with.

As per the Error Reason - Microsoft is suggesting to try again later .. so perhaps they are having some sort of transient problem.

**Comment by abraunegg** [comment]
@thorrrr 
Please also note my location and timezone - will review any updates in the morning.

**Reply by thorrrr** [reply]
Hello 
I have tried 5 times over the past 8 hours same error as above.
But my OneDrive windows and MS status page shows no errors with One Drive.

One thing i cannot work out and some of my more experienced linux friends is the location of the config file.
Every other app the config is located in .config/appname/config file but there is none on the OneDrive directory.

I only see the file I created but when i run OneDrive --display-config i see the config file with my file appended to the bottom.
But how do i edit the config file?

I cannot get past your number 2 instructions

https://paste.centos.org/view/80ec584b

```
onedrive --display-config
onedrive version                             = v2.4.19
Config path                                  = /home/dale/.config/onedrive
Config file found in config path             = false
Config option 'sync_dir'                     = /home/dale/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     =
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 5
Config option 'monitor_fullscan_frequency'   = 12
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               =
Config option 'azure_ad_endpoint'            =
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   =
Config option 'force_http_2'                 = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = true
sync_list contents:
# sync_list supports comments
#
# The ordering of entries is highly recommended - exclusions before inclusions
#
# Exclude temp folders under Documents
!Documents/temp*
# Exclude my secret data
!/Secret_data/*
#Exlude Following Folders
!/Videos/*
!/Pictures/*
!/Documents/*
!/Public/*
!/E-Books/*
!/Backup/*
!/Amex Complaint Docs/*
!/Apps/*
!/From Dell SD Card/*
!/Desktop/*

#Includes

# Include Documents folder
Documents/eBay Restructure
# Include all PDF documents
Documents/*.docx
# Include these names if they match any file or folder
PhraseExpress
Screenshots
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```

**Comment by abraunegg** [comment]
@thorrrr 

> Every other app the config is located in .config/appname/config file but there is none on the OneDrive directory.

Your *default* configuration folder will be: `~/.config/onedrive/`

By default, this folder will contain zero items. When using the application *defaults* - no 'config' file is needed. 

If you look at line 3 and 4 of your `--display-config` output:
```
Config path                                  = /home/dale/.config/onedrive
Config file found in config path             = false     <---- This is correct .. you have no 'config' file .....
```

If you want to put a 'config' file in this directory - for your specific usage:
```
wget https://raw.githubusercontent.com/abraunegg/onedrive/master/config -O /home/dale/.config/onedrive/config
```
This will put a 'config' file in that location which you can edit.

This is also detailed and explained here: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#configuration

If you want to then create a 'sync_list' file - to control what is synchronized, create this file in `~/.config/onedrive/`

This is also detailed and explained here: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a-selective-sync-via-sync_list-file

Once you have configured everything and have things running, your configuration will contain files like this:

```
drwx------  2 alex alex  4096 Aug 11 06:39 .
drwx------ 14 alex alex  4096 Aug 11 04:30 ..
-rw-rw-r--  1 alex alex  1707 Aug 11 06:38 config
-rw-r--r--  1 alex alex 32768 Aug 11 04:02 items.sqlite3
-rw-------  1 alex alex   874 Aug 11 04:02 refresh_token
-rw-rw-r--  1 alex alex   611 Aug 11 06:39 sync_list
```


> I have tried 5 times over the past 8 hours same error as above.
>But my OneDrive windows and MS status page shows no errors with One Drive.

If you are getting 500 internal errors from Microsoft - this is not something that can be assisted with ... that is a Microsoft issue.

Suggestion:
```
onedrive --logout
onedrive --reauth
```

Perhaps there is something up with your account / access / token

**Comment by thorrrr** [comment]
HI

Mine looks like this currently i did the sync_list this morning  as you can see from the link i sent you https://paste.centos.org/view/80ec584b 
But will try what you said again 
```
drwx------. 1 dale dale  74 Aug 10 13:10 .
drwxr-xr-x. 1 dale dale 834 Aug 10 12:54 ..
-rw-------. 1 dale dale 374 Aug 10 07:30 refresh_token
-rw-r--r--. 1 dale dale 567 Aug 10 11:24 sync_list
-rw-------. 1 dale dale  28 Aug 10 10:07 .sync_list.hash
```

**Comment by thorrrr** [comment]
HI 
I have tried l
```
onedrive --logout
onedrive --reauth
```
![image](https://user-images.githubusercontent.com/8170673/184025084-0ba80972-32bc-471d-b582-76fbaeafa33e.png)

I have only got 1 folder called Documents with nothing in it 


I run onedrive --synchronize --verbose
```
ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 500 (Internal Server Error)
  Error Reason:     Failed to get a coherent view of changes. Try again later.
  Error Timestamp:  2022-08-10T21:52:21
  API Request ID:   d41c0bcb-9608-4943-bc63-904cbd881de0
  Calling Function: applyDifferences()
```

**Comment by abraunegg** [comment]
@thorrrr 
Can you post the *full* log & output when using *this* specific command:

```
onedrive --synchronize --verbose --resync
```

Posting just the last error is not helpful.

Potentially something in the DB is corrupted with a bad /delta path.

Additionally - you are also using Fedora 36 packages - running client version v2.4.19 ....

If the above suggested command does not work and generates the same error, you will need to *upgrade* your client by building the client from source, or, potentially even downgrade your client to v2.4.18 to see if your actual issue is impacted by https://github.com/abraunegg/onedrive/pull/2017 (doubtful ... but ...)

```
sudo dnf downgrade onedrive
```

**Reply by thorrrr** [reply]
file:///home/dale/Documents/dale.onedrive.log

**Comment by thorrrr** [comment]
[dale.onedrive.LOG](https://github.com/abraunegg/onedrive/files/9306868/dale.onedrive.LOG)

**Comment by abraunegg** [comment]
@thorrrr 
You are not running this command as requested:
```
onedrive --synchronize --verbose --resync
```

Copy and paste that into your SSH or terminal session.

**Reply by thorrrr** [reply]
Yes i did i copied last time here is my terminal history
![Screenshot from 2022-08-11 08-22-24](https://user-images.githubusercontent.com/8170673/184083747-8df76fba-b1af-403c-bf37-193eba4c794b.png)

**Reply by abraunegg** [reply]
@thorrrr 
your log file is not correct for that command ......

**Comment by thorrrr** [comment]
Well i enabled logging this folder was blank before i started.
I then ran what you see and it created that log but i will run it again

**Reply by abraunegg** [reply]
@thorrrr 
If you want to enable logging - please read: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#client-activity-log to configure the logging directory & set group permissions.

Once done, you have 2 options to enable logging:

**Option 1: Edit the configuration file 'config'**
```
# disable_upload_validation = "false"
enable_logging = "true"
# force_http_11 = "false"

```

**Option 2: Use the --enable-logging flag**
```
onedrive --synchronize --verbose --resync --enable-logging
```

Without doing this .. logging to a separate file is *not* enabled.

**Reply by thorrrr** [reply]
running option 2 now

**Reply by thorrrr** [reply]
[dale.onedrive.LOG](https://github.com/abraunegg/onedrive/files/9307036/dale.onedrive.LOG)

**Comment by abraunegg** [comment]
@thorrrr 
OK .. thanks for the updated debug log. So what is going on:
```
......
2022-Aug-11 08:43:45.7296749	Processing 293 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:43:45.7364603	Skipping item - excluded by sync_list config: Videos
...
2022-Aug-11 08:43:47.0976559	Processing 219 OneDrive items to ensure consistent local state due to sync_list being used
...
2022-Aug-11 08:43:47.1094322	Skipping item - excluded by sync_list config: Documents/Excel Files
2022-Aug-11 08:43:49.0414139	Processing 211 OneDrive items to ensure consistent local state due to sync_list being used
...
2022-Aug-11 08:43:49.0525925	Skipping item - excluded by sync_list config: Documents/Excel Files
2022-Aug-11 08:43:50.2930453	Processing 223 OneDrive items to ensure consistent local state due to sync_list being used
```
This is all 100% normal operation.

Now this is where this is not normal:
```
2022-Aug-11 08:43:52.727105	Skipping item - excluded by sync_list config: Documents/Excel Files
2022-Aug-11 08:43:55.5288868	Processing 719 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:43:55.5373976	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:43:58.4821336	Processing 749 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:43:58.4896179	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:44:00.9307519	Processing 650 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:44:00.9383058	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:44:04.0697398	Processing 873 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:44:04.0776794	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:44:06.4375129	Processing 526 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:44:06.4478694	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:44:08.4611191	Processing 456 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:44:08.4687015	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:44:11.5064079	Processing 587 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:44:11.5142397	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:44:13.4002625	Processing 395 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:44:13.4093547	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:44:16.0161048	Processing 634 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:44:16.0238232	Skipping item - excluded by sync_list config: Documents/Archive
2022-Aug-11 08:44:19.1703301	Processing 753 OneDrive items to ensure consistent local state due to sync_list being used
2022-Aug-11 08:44:19.1813704	Skipping item - excluded by sync_list config: Documents/Archive
```

You see how the Processing XXX value jumps from ~200 items up to 500+ then bounces around 700+ ?

This is the OneDrive API being broken. The API specification is that 'changes' from OneDrive are sent in approx lots of 200. Never ever so many.

So - in the /delta query that the client uses - the client utilises the 'next' set of data as provided by the API - and - it appears that this is now horribly broken somewhere .. and this is something that needs to be reported to Microsoft.

Given they have been touting a new OneDrive release in the last few days - I suspect they have gone and broken something.

As this is a OneDrive Personal Account, I may be able to replicate this scenario (unsure - it might even be geographic specific) - but once I log the fault with Microsoft I will try and replicate, now that I know what the potential trigger is.

Logged with Microsoft: https://github.com/OneDrive/onedrive-api-docs/issues/1620

**Reply by thorrrr** [reply]
OK
So not my fault :)
I do have Office 365 for familly i guess that still counts as a personal onedrive account?
So shall i just wait until it is fixed?

**Reply by abraunegg** [reply]
I will try and replicate - if I can, this is good ...

If I cant ... then you will need to raise a separate case with Microsoft

>I do have Office 365 for familly i guess that still counts as a personal onedrive account?

The OneDrive API responds with the following as your account type:
```
2022-Aug-11 08:32:30.3344737	Account Type: personal
```

So - this is 100% a 'personal' account type for OneDrive operational purposes.

**Reply by thorrrr** [reply]
Thank you all your help, looking forward to trying the GUI when i get it working

**Comment by abraunegg** [comment]
@thorrrr 
I have tried to replicate this issue without success - so this leaves 3 options as to your problem:
1. This is an actual API issue specific to 'Office 365 for Family' accounts
2. This is an actual API issue specific to your geographic region
3. This is an actual API issue specific to account data held on OneDrive - potentially caused by 'rclone' to begin with.

Now .. all of the increases in the API responses appear to be happening when the API begins querying this online folder of yours: `Documents/Archive`

I am wondering if there is something 'online' that is causing this loop|redirection to occur. 

Happy to investigate this further with you, however this will require sharing contact details offline and doing an online meeting where screen sharing can be done - I think this is the only viable way here to potentially understand what is going on with your account online and potentially remediate this as I feel that this is less likely to be an API issue - rather than something with your folder structure online.

If you can connect offline using the support email address that would be greatly appreciated.

**Comment by thorrrr** [comment]
HI 
Sure when is good for you ?
Do you want me to contact you on support@mynas.com.au

**Reply by abraunegg** [reply]
Correct

**Reply by thorrrr** [reply]
HI i have emailed you twice have you received them?

**Reply by abraunegg** [reply]
Nothing has been received .. potentially this has been trashed by spam filters.

Please send a *new* email, with the following subject line: Linux OneDrive Discussion #2085

**Reply by thorrrr** [reply]
Sent again

**Reply by abraunegg** [reply]
Received & responded - will connect offline.
